pkg/provisioner/ironic/delete_test.go

```diff
@@ -190,3 +219,100 @@ func TestDelete(t *testing.T) {
 		})
 	}
 }
+
+func TestDeleteWithPowerOff(t *testing.T) {
+
+	nodeUUID := "33ce8659-7400-4c68-9535-d10766f07a58"
+
+	cases := []struct {
+		name        string
+		ironic      *testserver.IronicMock
+		inspector   *testserver.InspectorMock
+		priorErrors int
+
+		expectedPowerState string
+		expectedError      string
+	}{
+		{
+			name: "test-power-change-without-errors",
+			ironic: testserver.NewIronic(t).Node(
+				nodes.Node{
+					UUID:           nodeUUID,
+					ProvisionState: "active",
+					Maintenance:    true,
+					PowerState:     powerOn,
+				},
+			).WithNodeStatesPowerUpdate(nodeUUID, http.StatusAccepted),
+			priorErrors:        0,
+			expectedPowerState: "power off",
+			expectedError:      "failed to remove host",
```

Comment:
Since this test does not focus on delete action itself, this kind of expectedError sounds a little bit strange. It's better to mock the call to Delete to make it going on without errors.

Reply:
Done.